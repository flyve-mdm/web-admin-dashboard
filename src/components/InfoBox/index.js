import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

const InfoBox = ({to, icon, count, name}) => {
    return (
        <Link to={to}>
            <div className="info-box">
                <span className="content-box">
                    {count}
                </span>
                <span className={'icon-box ' + icon} />
                <span className="title-box">
                    { name.toUpperCase() }
                </span>
            </div>
        </Link>
    )
}

InfoBox.propTypes = {
    to: PropTypes.string.isRequired,
    icon: PropTypes.string.isRequired,
    count: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string
    ]).isRequired,
    name: PropTypes.string.isRequired,
}

export default InfoBox