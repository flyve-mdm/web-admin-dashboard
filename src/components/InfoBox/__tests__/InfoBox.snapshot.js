import React from 'react'
import renderer from 'react-test-renderer'
import { Router } from 'react-router-dom'
import history from '../../../shared/history'

import InfoBox from "../index"

describe('InfoBox', () => {
    test('renders InfoBox', () => {
        const component = renderer.create(
            <Router history={history}>
                <InfoBox 
                    name="test"
                    to="test/"
                    icon="filesIcon"
                    count={3}
                />
            </Router>
        )
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})