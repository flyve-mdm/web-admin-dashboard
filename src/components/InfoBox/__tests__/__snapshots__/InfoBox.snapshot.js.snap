// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBox renders InfoBox 1`] = `
<a
  href="test/"
  onClick={[Function]}
>
  <div
    className="info-box"
  >
    <span
      className="icon-box filesIcon"
    />
    <span
      className="content-box"
    >
      3
    </span>
    <span
      className="title-box"
    >
      TEST
    </span>
  </div>
</a>
`;
